fastlane/lib/fastlane/fastfile_parser.rb

```diff
@@ -0,0 +1,324 @@
+require 'parser'
+require 'parser/current'
+require 'pp'
+require "terminal-table"
+module Fastlane
+  class FastfileParser
+    attr_accessor :original_action
+
+    def to_s
+      "REMOVE_IT"
+    end
+
+    def lines
+      @lines ||= []
+    end
+
+    def lanes
+      @lanes ||= []
+    end
+
+    def actions
+      @actions ||= []
+    end
+
+    def counters
+      errors = lines.select { |key| key[:state] == :error }.length
+      deprecations = lines.select { |key| key[:state] == :deprecated }.length
+      infos = lines.select { |key| key[:state] == :infos }.length
+      { errors: errors, deprecations: deprecations, infos: infos, all: errors + infos + deprecations }
+    end
+
+    def data
+      { lanes: @lanes,  notices: @lines, actions: @actions }
+    end
+
+    def bad_options
+      [:use_legacy_build_api]
+    end
+
+    def redacted
+      rc = ""
+      rc = @content
+      self.class.secrets.each do |s|
+        rc.gsub!(s, "#" * s.length)
+      end
+      rc
+    end
+
+    # rubocop:disable PerceivedComplexity
+    def fake_action(*args)
+      # Suppress UI output
+
+      return_data = { args: args.first }
+      return return_data if args.length <= 0
+      return return_data if @original_action.nil?
+      UI.important("ACTION: #{@original_action.inspect} 1") if $verbose
+      UI.important("PARAMS: #{args.inspect}") if $verbose
+      a = Fastlane::Actions.action_class_ref(@original_action.to_sym)
+      a = find_alias(@original_action.to_sym) unless a
+      return return_data unless a
+
+      # Get the importet file
+      out_channel = StringIO.new
+      $stdout = out_channel
+      $stderr = out_channel
+      if @original_action.to_s == "import_from_git"
+        begin
+        fl = Fastlane::FastFile.new
+        fl.runner = Runner.new
+        path = fl.import_from_git(url: args.first[:url], branch: args.first[:branch] || "HEAD", return_file: true)
+
+        actions_path = File.join(path, 'actions')
+        Fastlane::Actions.load_external_actions(actions_path) if File.directory?(actions_path)
+
+        icontent = File.read("#{path}/fastlane/Fastfile")
+        fl_parser = FastfileParser.new(content: icontent, filepath: @dirname, name: "Imported at #{@line_number}", change_dir: false, platforms: @platforms)
+        fl_parser.analyze
+        @content << "#########################################\n"
+        @content << "# BEGIN import_from_git at: #{@line_number} BEGIN\n"
+        @content << "#########################################\n"
+        @content << icontent
+        @content << "#########################################\n"
```

Comment:
That's a lot of code just to import from Git. How about we don't support this here, but just show a message that only the local Fastfile will be linted, with instructions on how to also lint/test the Fastfile that should be imported?